src/server/server.ts

```diff
@@ -155,6 +155,38 @@ namespace ts.server {
 
     var logger = createLoggerFromEnv();
 
+    let pending: string[] = [];
+    function queueMessage(s: string) {
+        pending.push(s);
+        if (pending.length === 1) {
+            drain();
+        }
+    }
+
+    function drain() {
+        Debug.assert(pending.length > 0);
+        writeBuffer(new Buffer(pending[0], "utf8"), 0);
+    }
+
+    function writeBuffer(buffer: Buffer, offset: number) {
+        const toWrite = buffer.length - offset;
+        fs.write(1, buffer, offset, toWrite, undefined, (err, written, buffer) => {

```

Comment:
We are retrying instead of handling the error anyway, it may only be used for printing error messages I think.